pkg/api/registry/registry.proto

```diff
@@ -22,61 +22,44 @@ message Destination {
     uint32 weight = 2;
 }
 
-message NetworkServiceManager {
-    string name = 1;
-    string url = 2;
-    google.protobuf.Timestamp expiration_time = 3;
-    string state = 4;
-}
-
 message NetworkServiceEndpoint {
     string name = 1;
     string payload = 2;
     string network_service_name = 3;
-    string network_service_manager_name = 4;
-    map<string, string> labels = 5;
-    string state = 6;
-
+    map<string, string> labels = 4;
+    string state = 5;
     // Endpoint callback address, could be any of unix/tcp/callback schema ones.
-    string url = 7;
-}
-
-message FindNetworkServiceRequest {
-    string network_service_name = 1;
-}
-
-message FindNetworkServiceResponse {
-    string payload = 1;
-    NetworkService network_service = 2;
-    map<string, NetworkServiceManager> network_service_managers = 3;
-    repeated NetworkServiceEndpoint network_service_endpoints = 4;
+    string url = 6;
+    google.protobuf.Timestamp expiration_time = 7;
 }
 
 message NSERegistration {
     NetworkService network_service = 1;
-    NetworkServiceManager network_service_manager = 2;
+    NetworkServiceEndpoint network_service_manager = 2;
     NetworkServiceEndpoint network_service_endpoint = 3;
 }
 
-message RemoveNSERequest {
-    string network_service_endpoint_name = 1;
-}
-
 service NetworkServiceRegistry {
-    rpc RegisterNSE (NSERegistration) returns (NSERegistration);
-    rpc BulkRegisterNSE (stream NSERegistration) returns (stream NSERegistration);
-    rpc RemoveNSE (RemoveNSERequest) returns (google.protobuf.Empty);
+    rpc Register (NSERegistration) returns (NSERegistration);
+    rpc Monitor (NetworkServiceSelector) returns (stream NSERegistration);
+    rpc Remove (NetworkServiceSelector) returns (google.protobuf.Empty);
+    rpc Update (NSERegistration) returns (NSERegistration);
 }
 
 service NetworkServiceDiscovery {
-    rpc FindNetworkService (FindNetworkServiceRequest) returns (FindNetworkServiceResponse);
+    rpc FindNetworkService (NetworkServiceSelector) returns (stream NSERegistration);
 }
 
-message NetworkServiceEndpointList {
-    repeated NetworkServiceEndpoint network_service_endpoints = 1;
+enum NetworkServiceKind {
```

Comment:
Why do we need Kind here?

Reply:
https://github.com/networkservicemesh/api/pull/41#discussion_r434697556